module_utils/ansible_nmstate.py

```diff
@@ -64,7 +64,10 @@ def run(self):
         action()
 
     def apply_partial_interface_state(self, interface_state):
-        interfaces = [interface_state]
+        if isinstance(interface_state, list):
```

Comment:
What is the reason this func is called with two different types? (not that nice).

Reply:
It is needed for the modules that do not yet support aggregate. AFAIK this is the way to overload methods in Python.

Reply:
The proper way to overload is to inherit and re-define.